lib/dogapi/common.rb

```diff
@@ -210,14 +210,17 @@ def Dogapi.find_datadog_host
   @@hostname = nil
 
   def Dogapi.find_localhost
-    unless @@hostname
-      out, status = Open3.capture2('hostname', '-f', err: File::NULL)
-      # Get status to check if the call was successful
-      raise SystemCallError, 'Could not get hostname with `hostname -f`' unless status.exitstatus.zero?
-      @@hostname = out.strip
-    end
-  rescue SystemCallError
-    @@hostname = Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
+    @@hostname if @@hostname
+    out, status = Open3.capture2('hostname', '-f', err: File::NULL)
+    unless status.exitstatus.zero?
+      begin
+        out = Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
+      rescue SocketError
+        out, status = Open3.capture2('hostname', '-s', err: File::NULL)
```

Comment:
Do we actually want to use `hostname -s`? Or should we use just `hostname`? The reason why we bumped chef-handler was precisely because of a `hostname -s` warning (see #57)

Reply:
Good catch. I wasn't aware that Windows didn't support that flag. I used `-s` because on MacOS `-f` is the default, so on those if `hostname -f` fails, `hostname` will also fail. We have two options:
- Change the `hostname` call we do depending on the platform
- Remove the `-s` and assume that on Mac we will have one less error recovery option.
Given this error path should be uncommon (our two other ways to get the hostname must fail), I will go with the second option to not increase the code complexity.

Reply:
I like the second option more too, we can always add complexity later for macOS if we need to